src/Analyzers/CSharp/Tests/RemoveUnusedParametersAndValues/RemoveUnusedValueAssignmentTests.cs

```diff
@@ -10092,4 +10088,71 @@ void M()
             }
             """);
     }
+
+    [Fact]
```

Comment:
Consider work items

Reply:
Not sure where the related issue on GitHub is. The one I could find is https://github.com/dotnet/roslyn/issues/77258 but it's unrelated to ref-structs and won't be fixed by this PR.

Reply:
Not a problem!  Btw, do you know what would fix 77258?

Reply:
#77258 feels tricky. If accurate results are needed then I'm guessing some sort of interprocedural analysis may be needed?

```
int[] a = new int[5];
MyStruct m = new MyStruct(a);

// Just by looking at this, we can't tell whether or not this assignment is unnecessary
// So, either we go with false positives, or false negatives.
// Or, we go with interprocedural analysis that tries to understand exactly the indexer logic does and tries to relate it to the surrounding code, which seems very complicated (not even sure if possible to achieve).
m[0] = 1;

Console.WriteLine(a[0]);
```

Reply:
> If accurate results are needed then I'm guessing some sort of interprocedural analysis may be needed?

Intraprocedural analysis isn't safe to do and would represent at best a point in time snapshot. It's notably also "impossible" to do for cross assembly due to the existence of reference assemblies.

We would require some new attribute or construct that represents the necessary concept (which is most functionally similar to a `Pure` like annotation) and have it documented as a breaking change to remove from a method once it's been placed on it.

In general any property, indexer, or method may have side effects or involve indirect mutations; even if these members are annotated `readonly`.

Reply:
The question here is if we can make the field read-only. The field can be readonly if the user is only access read-only members off of it, as those won't cause copies to happen.

Reply:
> or involve indirect mutations; even if these members are annotated readonly.

An indirect mutation is fine. This analyzer is just asking if the field can be readonly.  It can be, while still making indirect mutations. 

Note: the fail case here is to not offer anything at all. So it's a narrow whitelist of cases we offer the feature for. And it's a very popular and widely used analyzer/fixer with widespread usage. We want it to still be useful in th those whitelisted cases, without accidentally having false positives.

Reply:
Ah, gotcha. The comment was confusing to me and I understood it as relating to `IDE0059` (unnecessary assignment of a value). It sounds like you're saying its rather in relation to `IDE0044` (add readonly modifier), `IDE0250` (struct can be made readonly), or `IDE0251` (member can be made readonly)

Reply:
Yup!  There are multiple features affected here, all with different desired behaviors :-)


"Make member read-only" wants to offer in all cases where it is totally safe to make readonly (so where it would not cause errors, or different runtime behavior (like a copy)). 

"Remove unnecessary assignment" wants to inform the user of a full overwrite of a variable (local, parameter) that won't be observed (either because there no following reads, or because there is another full write that follows). 

Different features. Different goals.  Subtle interactions since they use some common helpers which we need to be more vigilant about :-)